srtcore/api.cpp

```diff
@@ -370,10 +473,20 @@ int CUDTUnited::newConnection(const SRTSOCKET listen, const sockaddr_any& peer,
       return -1;
    }
 
-   enterCS(m_IDLock);
-   ns->m_SocketID = -- m_SocketIDGenerator;
-   HLOGF(mglog.Debug, "newConnection: generated socket id %d", ns->m_SocketID);
-   leaveCS(m_IDLock);
+   try
+   {
+       ns->m_SocketID = generateSocketID();
+   }
+   catch (CUDTException& e)
+   {
```

Comment:
`const CUDTException& e`

Reply:
Mind tho' that there are more cases like that in this file. Might need some refax to fix them all.